gammapy/spectrum/flux_point.py

```diff
@@ -596,11 +596,13 @@ class FluxPointEstimator(object):
     """
 
     def __init__(self, obs, groups, model):
-        self.obs = obs
+        self.obs = obs.copy()
```

Comment:
Why copy here? That's very expensive and shouldn't be needed, no?

Reply:
Based on a suggestion by Johannes, the way the procedure works now is by overwriting obs.on_vector.quality to indicate the good bins. Obviously we don't want to overwrite the array in the original observation so a copy is passed. The alternative is to leave obs intact and pass a separate array.

Reply:
@joleroi - Can this copy of obs or modification of it's quality data members be avoided?

The way I envision it for the `MapFit` is that it doesn't modify any of the data objects passed to it (e.g. maps, PSF, EDISP).

I guess working that way would also be nice here, but it's not easy to achieve?

Maybe making the copy is the best thing for now?
If so, please at least leave a comment above this line explaining why a copy is made, i.e. that the object is modified during fitting.

Reply:
Regarding passing a spectrum observation list, it seems strange that that is an acceptable input to SpectrumEnergyGroup, since in principle each observation could have its own obs.e_reco . I'm assuming in practice that never happens, in which case we could simply take the first element of the list if a list is passed. Would that be acceptable or do we need to do a check? If we allow observations with different e_reco we may have to copy the whole list (or pass a list of arrays, see above).

Reply:
I have no objection to including a data attribute in SpectrumEnergyGroup rather than modifying obs. It would be worth asking Johannes why he wanted to modify obs in the first place.